electron/scripts/child.js

```diff
@@ -87,25 +115,36 @@ function fillMedicationDropdown(json) {
 function fillMedicationTable(child_medications) {
 	let medication_table = document.getElementById('child_medication_table');
 	child_medications.sort(function(a,b) {
-		return (a.child_medication_end_date > b.child_medication_end_date) ? 1 : ((a.child_medication_end_date < b.child_medication_end_date) ? -1 : 0);
+		let date1 = a.child_medication_end_date; let date2 = b.child_medication_end_date;
+		console.log(date1,date2);
+		if (date1 < date2 || (date1 !== null && date2 === null)) {return 1;}
+		if (date1 > date2 || (date1 === null && date2 !== null)) {return -1;}
+		return 0;
 	}); 
 	for (let i=0; i<child_medications.length; i++) {
 		let row = document.createElement('TR');
 		let child_med_id = child_medications[i].id; 
+		row.id = "child_med_row_" + child_med_id;
 		// medication name
 		let table_str = '<td>' + child_medications[i].medication_english_name + '</td>';
 		// dosages
 		table_str += '<td>' + child_medications[i].dosage1 + '</td><td>' + child_medications[i].dosage2 + '</td><td>' + child_medications[i].dosage3 + '</td>';
 		// start/stop dates
 		table_str += '<td>' + child_medications[i].child_medication_start_date + '</td>';
-		table_str += '<td><input type="text" class="datepicker" id="medication_end_date_' + child_med_id + '" value="' + child_medications[i].child_medication_end_date + '" size="9"></td>';
+		let end_date_str = (child_medications[i].child_medication_end_date === null) ? "" : child_medications[i].child_medication_end_date;
+		//table_str += '<td><input type="text" class="datepicker" id="medication_end_date_' + child_med_id + '" value="' + end_date_str + '" size="9"></td>';
+		table_str += '<td id="medication_end_date_' + child_med_id + '">' + end_date_str + '</td>';
 		// ended checkbox
 		table_str += '<td><input type="checkbox" id="medication_finished_' + child_med_id + '" onchange=\'endChildMedication("' + child_med_id + '",this)\'';
```

Comment:
This table construction code looks hacky. Maybe make a class for the rows in the table and then just iterate through the elements of your class? also, using the DOM API rather than just concatenating HTML strings together seems like a more robust approach. Maybe take a look at "constructTable()" in flagged_notes.js. In that I'm just iterating through a list of objects and making rows and tables using the DOM API.